resource_elasticsearch_odfe_role.go

```diff
@@ -0,0 +1,308 @@
+		Delete: resourceElasticsearchOdfeRoleDelete,
+		Schema: map[string]*schema.Schema{
+			"role_name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"cluster_permissions": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"index_permissions": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"index_patterns": &schema.Schema{
+							Type:     schema.TypeSet,
+							Optional: true,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+						},
+						"fls": &schema.Schema{
+							Type:     schema.TypeSet,
+							Optional: true,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+						},
+						"masked_fields": &schema.Schema{
+							Type:     schema.TypeSet,
+							Optional: true,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+						},
+						"allowed_actions": &schema.Schema{
+							Type:     schema.TypeSet,
+							Optional: true,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+						},
+					},
+				},
+			},
+			"tenant_permissions": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"tenant_patterns": &schema.Schema{
+							Type:     schema.TypeSet,
+							Optional: true,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+						},
+						"allowed_actions": &schema.Schema{
+							Type:     schema.TypeSet,
+							Optional: true,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+						},
+					},
+				},
+			},
+			"description": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		},
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+	}
+}
+
+func resourceElasticsearchOdfeRoleCreate(d *schema.ResourceData, m interface{}) error {
+	_, err := resourceElasticsearchPutOdfeRole(d, m)
+
+	if err != nil {
+		return err
+	}
+
+	name := d.Get("role_name").(string)
+	d.SetId(name)
+	return resourceElasticsearchOdfeRoleRead(d, m)
+}
+
+func resourceElasticsearchOdfeRoleRead(d *schema.ResourceData, m interface{}) error {
+	res, err := resourceElasticsearchGetOdfeRole(d.Id(), m)
+
+	if elastic7.IsNotFound(err) {
+		log.Printf("[WARN] OdfeRole (%s) not found, removing from state", d.Id())
+		d.SetId("")
+		return nil
+	}
+
+	if err != nil {
+		return err
+	}
+
+	d.Set("tenant_permissions", res.TenantPermissions)
+	d.Set("cluster_permissions", res.ClusterPermissions)
+	d.Set("index_permissions", res.IndexPermissions)
+	d.Set("description", res.Description)
+
+	return nil
+}
+
+func resourceElasticsearchOdfeRoleUpdate(d *schema.ResourceData, m interface{}) error {
+	_, err := resourceElasticsearchPutOdfeRole(d, m)
```

Comment:
There is no use to split this into 2 statements. err is not needed after the if.
```suggestion
	if _, err := resourceElasticsearchPutOdfeRole(d, m); err != nil {
	    return err
	}
```